hazelcast/src/main/java/com/hazelcast/core/Cluster.java

```diff
@@ -84,6 +84,18 @@
     Member getLocalMember();
 
     /**
+     * Promotes local lite member to data member.
+     * When this method returns both {@link #getLocalMember()} and {@link #getMembers()}
+     * reflects the promotion.
+     *
+     * @throws IllegalStateException if member is not a lite member
+     * @throws IllegalStateException if mastership claim in progress
+     * @throws IllegalArgumentException if local member cannot be identified as a member of the cluster
```

Comment:
Aha.. that case happens when master changes during lite member promotion. 

Suppose lite member sens request to master (Member-A). 
- Member-A processes the request, sends the response back but then crashes without being able to publish new member list.
- Member-B becomes the new master and lite member accepts its mastership. 

In this case, `updateMembers(view, master.getAddress(), master.getUuid())` call on lite member will fail, because master is changed. So, even though it receives a successful response to its request from the former master, its promotion is not applied to member list. That's why it fails there.